runtime/src/test/java/com/flipkart/flux/resource/StateMachineResourceTest.java

```diff
@@ -284,6 +284,105 @@ public void testPostScheduledEvent_withTriggerTimeInMilliSeconds() throws Except
 
     }
 
+    @Test
+    public void testPostEventUpdate_withNoEventData() throws Exception {
+        String stateMachineDefinitionJson = IOUtils.toString(this.getClass().getClassLoader().getResourceAsStream("state_machine_definition.json"));
```

Comment:
There could be cases where event data could be null , in case when they use dummy events to orchestrate their use case. Just check that the field is present, which could be empty., we can discuss this.